common/sketch/src/main/java/org/apache/spark/util/sketch/CountMinSketchImpl.java

```diff
@@ -348,4 +374,30 @@ public static CountMinSketchImpl readFrom(InputStream in) throws IOException {
 
     return new CountMinSketchImpl(depth, width, totalCount, hashA, table);
   }
+
+  @Override
+  public void writeExternal(ObjectOutput out) throws IOException {
+    try (ByteArrayOutputStream bos = new ByteArrayOutputStream()) {
+      this.writeTo(bos);
+      byte[] bytes = bos.toByteArray();
+      out.writeObject(bytes);
```

Comment:
bloom filters can get pretty big when cardinality is high (e.g. 500MB). It's best if we can avoid the extra copying.